user-guide/Advanced_Modules/Dashboards_app/Using_dashboard_feeds/Configuring_GQI_feeds/Configuring_an_external_data_source_in_a_query.md

```diff
@@ -47,6 +47,21 @@ Depending on how the script is configured, there can be additional configuration
 
 An ad hoc data source is represented as a class that implements predefined interfaces. The interfaces you can use are detailed below.
 
+From DataMiner 10.3.4/10.4.0 onwards, ad hoc data sources can retrieve data by means of DMS messages. To do so, the [*IGQIDataSource* interface](#igqidatasource) must implement the [*IGQIOnInit* interface](#igqioninit), of which the `OnInit` method can als be used to initialize a data source:
+
+```csharp
+OnInitOutputArgs OnInit(OnInitInputArgs args)
+```
+
+When passed to the `OnInit` method, `OnInitInputArgs` can contain the following new property:
```

Comment:
Can you maybe add a link here to the ### GQIDMS section below?
Then I think it's fine that these 2 additions are in different places.